File: 2364-count-number-of-bad-pairs/main.py
class Solution:
    def countBadPairs(self, nums: List[int]) -> int:
        
        n = len(nums)
        total_pairs = n * (n -1)// 2
        good_pairs = 0

        diff_count = defaultdict(int)

        for i, num in enumerate(nums):
            diff = num - i
            good_pairs += diff_count[diff]
            diff_count[diff] += 1

        return total_pairs - good_pairs

File: 2364-count-number-of-bad-pairs/main.rs
use std::collections::HashMap;

impl Solution {
    pub fn count_bad_pairs(nums: Vec<i32>) -> i64 {
        
        let n = nums.len();
        let total_pairs: i64 = (n * (n-1) / 2) as i64;   
        let mut good_pairs: i64 = 0;

        let mut diff_count: HashMap<i32, i32> = HashMap::new();

        for i in 0..n {
            let diff = nums[i] - i as i32;
            if let Some(&count) = diff_count.get(&diff) {
                good_pairs += count as i64;
            }
            *diff_count.entry(diff).or_insert(0) += 1;
        }

        total_pairs - good_pairs
    }
}
